### PR TITLE
[Distributed] IRGen: A couple of fixes for accessor

### PR DESCRIPTION
**Pull request by xedin**
*  Make accessor `thin` and require actor as (last) guaranteed argument (well typed actor argument)
* Construct pointer to a distributed thunk correctly (using async function pointer)
* Adjust continuation to expect `SwiftError *` in its throwing variant

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

File: include/swift/ABI/Executor.h [hunk 1]
```diff
@@ -214,7 +214,7 @@ struct AsyncContinuationTypeImpl<
   AsyncSignature<void(ArgTys...), /*throws=*/true>> {
 
   using type = SWIFT_CC(swiftasync) void(SWIFT_ASYNC_CONTEXT AsyncContext *,
-                                         SWIFT_CONTEXT void *);
+                                         SWIFT_CONTEXT SwiftError *);
```
Comment (ktoso) [inline include/swift/ABI/Executor.h:217]: oh, nice 👍

File: lib/IRGen/GenDistributed.cpp [hunk 2]
```diff
@@ -391,12 +392,12 @@ FunctionPointer DistributedAccessor::getPointerToMethod() const {
   auto signature = IGM.getSignature(fnType, fpKind.useSpecialConvention());
 
   auto *fnPtr =
-      IGM.getAddrOfSILFunction(Method, NotForDefinition,
-                               /*isDynamicallyReplaceable=*/false,
-                               /*shouldCallPreviousImplementation=*/false);
+    llvm::ConstantExpr::getBitCast(IGM.getAddrOfAsyncFunctionPointer(Method),
+                                   signature.getType()->getPointerTo());
```
Comment (ktoso) [inline lib/IRGen/GenDistributed.cpp:396]: ah so that's the "we need an async function pointer as function pointer" bit...

**Review by ktoso — APPROVED** [review]
Awesome, LGTM

File: lib/IRGen/GenDistributed.cpp (hunk 2 above)
Comment (xedin) [reply lib/IRGen/GenDistributed.cpp:396]: Yeah…

**Pull request merged**